internal/uvm/create_wcow.go

```diff
@@ -308,36 +476,29 @@ func CreateWCOW(ctx context.Context, opts *OptionsWCOW) (_ *UtilityVM, err error
 		return nil, errors.Wrap(err, errBadUVMOpts.Error())
 	}
 
-	doc, err := prepareConfigDoc(ctx, uvm, opts)
+	var doc *hcsschema.ComputeSystem
+	if opts.SecurityPolicyEnabled {
+		doc, err = prepareSecurityConfigDoc(ctx, uvm, opts)
+		log.G(ctx).Tracef("CreateWCOW prepareSecurityConfigDoc result doc: %v err %v", doc, err)
+	} else {
+		doc, err = prepareConfigDoc(ctx, uvm, opts)
+		log.G(ctx).Tracef("CreateWCOW prepareConfigDoc result doc: %v err %v", doc, err)
+	}
 	if err != nil {
 		return nil, fmt.Errorf("error in preparing config doc: %w", err)
 	}
 
-	if err := wclayer.GrantVmAccess(ctx, uvm.id, opts.BootFiles.ScratchVHDPath); err != nil {
-		return nil, errors.Wrap(err, "failed to grant vm access to scratch")
-	}
-
-	doc.VirtualMachine.Devices.Scsi = map[string]hcsschema.Scsi{}
-	for i := 0; i < int(uvm.scsiControllerCount); i++ {
-		doc.VirtualMachine.Devices.Scsi[guestrequest.ScsiControllerGuids[i]] = hcsschema.Scsi{
-			Attachments: make(map[string]hcsschema.Attachment),
-		}
-	}
-
-	doc.VirtualMachine.Devices.Scsi[guestrequest.ScsiControllerGuids[0]].Attachments["0"] = hcsschema.Attachment{
-
-		Path:  opts.BootFiles.ScratchVHDPath,
-		Type_: "VirtualDisk",
-	}
-
-	uvm.reservedSCSISlots = append(uvm.reservedSCSISlots, scsi.Slot{Controller: 0, LUN: 0})
-
 	err = uvm.create(ctx, doc)
 	if err != nil {
 		return nil, fmt.Errorf("error while creating the compute system: %w", err)
 	}
 
-	if err = uvm.startExternalGcsListener(ctx); err != nil {
+	if opts.SecurityPolicyEnabled {
+		err = uvm.startExternalGcsListener(ctx, windowsSidecarGcsHvsockServiceID)
+	} else {
+		err = uvm.startExternalGcsListener(ctx, gcs.WindowsGcsHvsockServiceID)
+	}
+	if err != nil {
```

Comment:
maybe this so we dont duplicate the same function call?
```suggestion
    gcsGUID := gcs.WindowsGcsHvsockServiceID
	if opts.SecurityPolicyEnabled {
		gcsGUID = windowsSidecarGcsHvsockServiceID
	}
	if err = uvm.startExternalGcsListener(ctx); err != nil {
```